synapse/storage/databases/main/registration.py

```diff
@@ -394,7 +394,9 @@ async def get_users_expiring_soon(self) -> List[Dict[str, Any]]:
             A list of dictionaries, each with a user ID and expiration time (in milliseconds).
         """
 
-        def select_users_txn(txn, now_ms, renew_at):
+        def select_users_txn(
+            txn: LoggingTransaction, now_ms: int, renew_at: int
+        ) -> List[Dict[str, Any]]:
```

Comment:
I'm always reluctant to use `Any`: it means the typechecker can't do anything with this data.

There's only one call site for this function. We could consider refactoring it to return `List[Tuple[str, int]]`.

(To be explicit: there's nothing wrong with this change; I just like empowering mypy as much as possible.)

Reply:
I have refactore it to use `List[Tuple[str, int]]`